src/serial_impl.rs

```diff
@@ -0,0 +1,52 @@
+//! Implementation of [`Serial`](https://docs.rs/embedded-hal/0.2.1/embedded_hal/serial/index.html)
+
+
+use nb;
+use ::hal::serial::{Read, Write};
+use ::serial;
+
+/// Newtype around [`serial::SystemPort`] that implements the `embedded-hal` traits
+pub struct Serial(pub serial::SystemPort);
+
+impl Read<u8> for Serial {
+    type Error = serial::Error;
+
+    fn read(&mut self) -> nb::Result<u8, Self::Error> {
+        use std::io::Read;
+        let mut buffer = [0; 1];
+        let bytes_read = self.0.read(&mut buffer).unwrap();
```

Comment:
Why the `unwrap`? Can't we just pass the error through?

Reply:
I needed to convert the error. Fixed.